survey-results/Readme.md

```diff
@@ -0,0 +1,318 @@
+
+The word cloud ended up looking like this (link contains higher resolution image):
+
+[![Bubble chart showing most used crates](crates.svg)](crates.svg)
+
+### Where to Improve
+
+We then asked what areas these crates could use the most improvement. The resounding answer (and repeated throughout the survey) is more comprehensive documentation, tutorials, and guides/examples. Requests for better documentation and examples made up over half of the responses to this question, with 45% and 57% respectively!
+
+[![Bar chart showing areas of where to improve](what-to-improve.svg)](what-to-improve.svg)
+
+Many times throughout the survey there were responses requesting the ability to do a thing, when there is in fact an excellent crate already available for that thing. There were also many responses where certain features were requested of particular named crates, when that feature already exists in said crate.
+
+This just re-enforces the need in our mind for better documentation and examples. I plan on revisiting the documentation and examples for my own crates because of these responses.
+
+The CLI-WG has a list of "Tracking Issues" in our repository for discussing solutions and problems releated to the working group. Because of the survey, we've opened up more issues to address concerns raised. 
+
+For this first section we've opened two issues:
+
+* [The "ideal guide" for writing CLI apps in Rust · Issue #6 · rust-lang-nursery/cli-wg · GitHub](https://github.com/rust-lang-nursery/cli-wg/issues/6)) 
+* [Improve ecosystem discoverability · Issue #11 · rust-lang-nursery/cli-wg · GitHub](https://github.com/rust-lang-nursery/cli-wg/issues/11)
+
+Example and Documentation will also be key areas of focus when contributing or maintaining crates.
+
+Platform Availability was in second place (with ~14%), hence we're determined to consider the big three platforms (Linux, Windows, and macOS) while discussing and creating any official solutions, or reviews from the CLI-WG.
+
+### Existing Gaps
+
+Next we asked what are the current gaps in the ecosystem when working on CLI applications, and what the respondent did about them. The results can be summed up into these bullets (not necessarily in order):
+
+* Better guides/examples for CLI application structure
+* (Cross Platform) Configuration File Management
+  * Better ergonomics of merging various levels of configs (global, project, CLI flags)
+* Better ergonomics of File Paths
+* Release/Packaging guides/examples
+  * More Docs/Tooling around Binary Releases
+* Clean readline implementation, problems with each current offering
+* Standardize generation / installation of manpages
+* Better CLI testing
+* Terminal Coloring (esp. Windows)
+* Better logging (or decrease current complexities)
+* Better documentation of crates
+* More libs to 1.0
+* Improve Error Handling
+* Discover ability of crates
+
+The raw word cloud looks like this (link contains higher resolution image): 
```

Comment:
I don't know how feasible it would be to cross-reference the responses of "write it myself" to what other thingy the users found lacking. This way we might get an interesting few highlights of points that people were more likely to write themselves?

Reply:
There are a lot of free form comments in the survey results with very good feedback. While the goal of this post was to summarize the data (and quantify stuff), we should totally go over the raw answers again at some point.